### PR TITLE
fix: target and grant schemas #1858

### LINKED ISSUE #1858: fhir-persistence-schema doesn't behave as expected.
**Describe the bug**
1. ibm-fhir-schematool doesn't specify the target schema to be granted permissions for oauth/jbatch.
2. ibm-fhir-schematool doesn't specify the target schema to be dropped for oauth/jbatch or fhirdata.
3. fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java doesn't allow schema names to be set.

**To Reproduce**

1. Create a postgres db
```
docker run -itd \
    --rm --name postgresfhir \
    --shm-size=256m \
    --sysctl net.core.somaxconn=256 \
    --sysctl net.ipv4.tcp_syncookies=0 \
    --hostname db \
    -e POSTGRES_USER=postgres \
    -e POSTGRES_PASSWORD=change-password \
    -e PGDATA=/var/lib/postgresql/data/pgdata \
    -e POSTGRES_DB=fhirdb \
    -p 5432:5432 \
    --privileged \
    -v "$PWD/my-postgres.conf":/etc/postgresql/postgresql.conf \
    postgres:12.5 postgres -c 'config_file=/etc/postgresql/postgresql.conf'
```

Be sure max_locks_per_transaction = 10000 is set.

2. Run the following
- `docker run --rm  --env ENV_TOOL_INPUT=$(cat persistence-onboard-example.json | base64) ibm-fhir-schematool:latest`
- `docker run --rm  --env ENV_TOOL_INPUT=$(cat persistence-offboard-example.json | base64) ibm-fhir-schematool:latest`

**Expected behavior**
The grant should work properly and the drop-schema should be dropped.

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by prb112**
closes #1858

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-persistence-schema/README.md [hunk 1]
```diff
@@ -275,6 +275,9 @@ For PostgreSQL:
 --db-type postgresql
 ```
 
+Alternatively, you can drop specific schemas with `--drop-schema-batch schema-name-to-drop` and
+`--drop-schema-oauth schema-name-to-drop`
```
Comment (lmsurpre) [inline fhir-persistence-schema/README.md:279]: It seems a little odd to me because we don't allow the user to specify the schema name during the create.

Comment (prb112) [reply fhir-persistence-schema/README.md:279]:
```
            case "--create-schema-oauth":
                this.createOauthSchema = true;
                if (nextIdx < args.length && !args[nextIdx].startsWith("--")) {
                    schema.setOauthSchemaName(args[nextIdx]);
                    i++;
                }
                break;

Comment (prb112) [reply fhir-persistence-schema/README.md:279]: if the next value is not a name, but a value it puts it as the oauth schema name, same for fhir, and batch

Comment (lmsurpre) [reply fhir-persistence-schema/README.md:279]: interesting.  i've always just created the schemas with `--create-schemas` which takes the fhir schema name from `--schema-name` property and uses the hardcoded values for the other two.  i do vaguely remember adding the alternative now and listing them under https://github.com/IBM/FHIR/tree/master/fhir-persistence-schema#alternative-setting-up-a-shared-db2-with-separate-schemas-for-each-tenant

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**